packages/Input/src/Input.js

```diff
@@ -38,33 +89,11 @@ const Input = React.forwardRef((props, ref) => {
     return <span className="form-input__icon">{props.icon}</span>;
   };
 
-  const {
-    a11yText,
-    clearIcon,
-    className,
-    icon,
-    isDisabled,
-    isReadOnly,
-    hasClearButton,
-    hasError,
-    onClear,
-    size,
-    ...moreProps
-  } = props;
-
-  if (moreProps.value || moreProps.value === "") {
-    delete moreProps.defaultValue;
-  } else {
-    delete moreProps.value;
-  }
-
```

Comment:
New approach is to use `isControlled`, defined above to more explicitly branch control.